python/cudf/cudf/_lib/spillable_buffer.pyx

```diff
@@ -0,0 +1,448 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+import collections.abc
+import pickle
+import time
+from threading import RLock
+from typing import Any, Tuple, Union
+
+import numpy
+
+import rmm
+
+from cudf.core.buffer import (
+    Buffer,
+    DeviceBufferLike,
+    get_ptr_and_size,
+    is_c_contiguous,
+)
+from cudf.utils.string import format_bytes
+
+from cython.operator cimport dereference
+from libc.stdint cimport uintptr_t
+from libcpp.memory cimport make_shared, shared_ptr, static_pointer_cast
+from libcpp.vector cimport vector
+
+
+cdef shared_ptr[void] create_expose_counter():
+    """Create an expose counter"""
+    return static_pointer_cast[void, int](
+        make_shared[int](42)
```

Comment:
OK having read all the code I see how this is working, and wow... this feels like a massive hack, but it is also genius :joy: Can we not just write a simple counter class that uses a lock, rather than maintaining a shared pointer to a magic number and use the reference count of that pointer? This works... it just feels like an incredible abuse of shared pointer reference counting. Kudos to you for the idea, though!

Reply:
Or, another alternative is to just create an empty `shared_ptr`. `std::shared_ptr<void> x{}` is valid C++, if Cython lets you express that it would also be OK. Not my favorite, but I think acceptable.

Reply:
The initial idea was to subclass `column_view` and have it referencing this shared pointer. However, now that we use `SpillLock`, we don't need a `std::shared_ptr` anymore